code/system/application/sensation/src/geolocation/geolocation.cpp

```diff
@@ -212,90 +212,120 @@ odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode Geolocation::body()
+        auto cov = m_ekf.getCovariance();
 
-        auto cov_eigenvalues = cov.eigenvalues();
+//        auto cov_eigenvalues = cov.eigenvalues();
 
         auto cov_diagonal = cov.diagonal();
 
-        auto cov_determinant = cov.determinant();
+//        auto cov_determinant = cov.determinant();
 
         std::cout   << getName() << "\t"
-                    << "Covariance Matrix = \n" << cov
-                    << "Eigenvalues = \n" << cov_eigenvalues
-                    << "Diagonal = \n " << cov_diagonal
-                    << "Determinand =   " << cov_determinant
+//                    << "\nCovariance Matrix = \n" << cov
+//                    << "\nEigenvalues = \n" << cov_eigenvalues
+                    << "\nDiagonal = \n " << cov_diagonal
+//                    << "\nDeterminand =   " << cov_determinant
                     << std::endl;
-*/
 
 
 
 
 
       // Build the proper GPS coordinates to send
       opendlv::data::environment::Point3 currentStateEstimation
-              (state.x(), state.y(), currentCartesianLocation.getZ());
-      opendlv::data::environment::WGS84Coordinate 
-          currentWGS84CoordinateEstimation =
-          gpsReference.transform(currentStateEstimation);
-      double heading = state.theta();
+              (state.x() + m_gpsToCoGDisplacement_x,
+               state.y() + m_gpsToCoGDisplacement_y,
+               currentCartesianLocation.getZ() + m_gpsToCoGDisplacement_z);
+
+//TODO: convert here x and y to get the position of the rear axle
+
+      opendlv::data::environment::WGS84Coordinate currentWGS84CoordinateEstimation =
+              gpsReference.transform(currentStateEstimation);
+      double heading = std::asin(std::sin(state.theta()));  // asin(sin(theta)) make sure that the heading is in [-pi,+pi]
+
+      double positionConfidence = calculatePositionConfidence();
+
+      double headingConfidence = calculateHeadingConfidence();
+
+      double speedConfidence = calculateSpeedConfidence();
+
+      double yawRate = state.theta_dot();
+
+      double yawRateConfidence = calculateHeadingConfidence();
+
+      double longitudinalAcceleration = 0.0;
+
+      double longitudinalAccelerationConfidence = 0.0;
+
 
       std::cout   << getName() << "\t" << "timestamp="
         << timestamp << "\t "
         << "\tlat=" << currentWGS84CoordinateEstimation.getLatitude()
-        << ", long=" << std::setprecision(19) 
+        << ", long=" << std::setprecision(19)
         << currentWGS84CoordinateEstimation.getLongitude()
         << ", theta=" << std::setprecision(19) << state.theta() << std::endl;
 
       // Send the message
-      opendlv::sensation::Geolocation geoLocationEstimation(
-          currentWGS84CoordinateEstimation.getLatitude(),
-          currentWGS84CoordinateEstimation.getLongitude(),
-          gpsReading.getAltitude(),
-          heading);
+
+      opendlv::sensation::Geolocation geoLocationEstimation(currentWGS84CoordinateEstimation.getLatitude(),
```

Comment:
You should not align indentation. Use double indent (4 spaces) when breaking a line.